indico/modules/events/layout/forms.py

```diff
@@ -162,8 +163,11 @@ class MenuUserEntryFormBase(IndicoForm):
     title = StringField(_('Title'), [DataRequired()])
     is_enabled = BooleanField(_('Show'), widget=SwitchWidget())
     new_tab = BooleanField(_('Open in a new tab'), widget=SwitchWidget())
-    access = IndicoEnumRadioField(_('Visibility'), enum=MenuEntryAccess,
-                                  description=_('Specify who can see this menu item.'))
+    acl_entries = PrincipalListField(_('Visibility'), event=lambda form: form.event, allow_groups=True, allow_event_roles=True, allow_registration_forms=True, allow_speakers=True)
+
+    def __init__(self, *args, **kwargs):
+        self.event = kwargs.pop('event')
```

Comment:
This way of getting a mandatory `event` kwargs is pretty much a Python 2 leftover. Nowadays you can do it in a much nicer way:

```suggestion
    def __init__(self, *args, event, **kwargs):
        self.event = event
```